packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -203,78 +204,47 @@ const EDiscovery = WebexPlugin.extend({
    * @param {UUID} reportId - Id of the report being requested
    * @param {Object} options - optional parameters for this method
-   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 90s
+   * @param {number} options.offset - start position from which to retrieve records
+   * @param {number} options.size - the number of records to retrieve
+   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 30s
    * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the content summary
    */
   getContentSummary(reportId, options) {
     if (!reportId) {
       throw Error('Undefined parameter');
     }
 
-    const contentSummaryDefaultOptions = {timeoutMs: 90000};
-
     // use spread operator to set default options
-    options = {...contentSummaryDefaultOptions, ...options};
-
-    let offset = 0;
-    const size = 100;
-    let contentSummary = [];
-
-    function getNextContentSummaryPage(plugin) {
-      return plugin.request({
-        method: 'GET',
-        service: 'ediscovery',
-        resource: `reports/${reportId}/contents/summary`,
-        qs: {offset, size},
-        timeout: options.timeoutMs
-      })
-        .then((res) => {
-          if (!plugin.contentSummaryBySpaceIdCache.has(reportId)) {
-            plugin.contentSummaryBySpaceIdCache.set(reportId, new Map());
-          }
-
-          for (let i = 0; i < res.body.length; i += 1) {
-            if (res.body[i] && res.body[i].spaceId) {
-              try {
-                // The results are written to the contentSummaryBySpaceIdCache since information for
-                // a space is likely to be accessed very frequently when decrypting activities
-                plugin.contentSummaryBySpaceIdCache.get(reportId).set(res.body[i].spaceId, res.body[i]);
-              }
-              catch (error) {
-                plugin.logger.error(`Error adding ${res.body[i].spaceId} to contentSummaryBySpaceIdCache: ${error}`);
-              }
-            }
-            else {
-              plugin.logger.error('Error adding undefined space to contentSummaryBySpaceIdCache');
-            }
-          }
-
-          contentSummary = contentSummary.concat(res.body);
-          if (res.body.length < size) {
-            return contentSummary;
-          }
-          offset += size;
-
-          return getNextContentSummaryPage(plugin);
-        });
-    }
+    options = {...defaultOptions, ...options};
+
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: `reports/${reportId}/contents/summary`,
+      qs: {offset: options.offset, size: options.size},
+      timeout: options.timeoutMs
+    })
+      .then((res) => {
+        const cachePromise = this._writeSpacesToContentSummaryBySpaceIdCache(reportId, res.body);
 
-    return getNextContentSummaryPage(this);
+        return Promise.all([cachePromise])
```

Comment:
Does ._writeSpacesToCon... return an array of promises?

Reply:
It doesn't appear to, so wondering about the promise.all instead of chaining on the then

Reply:
Good point Andrew, I'll change to a then